inbm-v5.md

```diff
@@ -0,0 +1,335 @@
+         - update logging and state files
+         - send intermediate results to `aggregator` for reporting
+         - trigger reboot of platform if applicable
+      - _OS updater_: Perfomrs OS update related tasks like:
+         - check applicability, e.g. checks available disk space
+         - download OS image file and perform signature checks if applicable
+         - invoke OS update tool based on underlying OS type/distribution.
+         - update logging and state files
+         - send intermediate results to `aggregator` for reporting
+         - trigger reboot of platform if applicable
+      - _Application updater_: Perfomrs application update related tasks like:
+         - check applicability, e.g. checks available disk space
+         - invoke underlying OS distributions `package manager` to perfomr the required installation tasks.
+         - update logging and state files
+         - send results to `aggregator` for reporting
+
+1. #### telemerty-reporter
+   - **Function**: Thread performing basic platform telemetry collection and reporting
+   - **Main Tasks**: Basic plaform telemetry being collected by `telemetry-reporter` can be catogarized as `static` and `dynamic`
+      - _Static_: Information that remains same for the most part of the a devices life cycle (e.g. UUID, Serial number etc) or only changes on updates (e.g. Firmware version, OS version etc)
+      - _Dynamic_: Information which constantly changes and is ideal to be plotted on a `time-sereis` database (e.g. CPU usage, memory usage etc)
+
+1. #### aggregator
+   - **Function**: Thread reporting status to `inbm-daemon`
+   - **Main Tasks**: This thread performs the task of combining and reporting information from platform to be sent to the cloud or `inbc` tool:
+      - _Event_: information that needs to be sent during the update process to indicat the progress
+      - _Status_: final result of an update
+      - _Telemetry_: `static` and `dynamic` telemetry
+
+Data Flow
+---------
+
+.. 
+   Guidelines:
+   1. Break down the system’s data flow into stages.
+   2. Highlight how components interact and communicate.
+
+Workflow Stages
+~~~~~~~~~~~~~~~
+
+1. **Stage 1: Input**:
+   - Example: Resource requests submitted via APIs or command-line tools.
+
+2. **Stage 2: Processing**:
+   - Example: Scheduling, allocation of resources, or preprocessing of data.
+
+3. **Stage 3: Output**:
+   - Example: Execution of workloads, generation of results, or API responses.
+
+Extensibility
+-------------
+
+.. 
+   Guidelines:
+   1. Explain how the system can be customized or extended.
+   2. Include modular integration points or plugin options.
+
+- **Adding Components**: [Explain how to add new modules or features].
+- **Customizing Workflows**: [Describe how to adapt the system to unique requirements].
+- **Integration Points**: [List APIs or interfaces for external integrations].
+
+Deployment
+----------
+
+[Content of Deployment]
```

Comment:
Would be similar to existing Turtle Creek. Possibilities:
* Yocto recipe if desired
* .debs maybe with installer script
* Tiber packages
* `udm-bootstrap`

Would also mention installed vs. provisioned, and decommissioning.